content/guides/models/track/runs/multiple-runs-per-process

```diff
@@ -0,0 +1,147 @@
+
+---
+
+## Example: Maintaining a Primary (Long-Lived) Run + Sub-Runs
+
+Suppose you want to run a **primary task** (like model training) in “Run A,” which stays open for the script’s entire lifespan. Meanwhile, you periodically spawn **short-lived “secondary” runs** (Run B1, B2, etc.) for tasks like evaluation—**without** finishing Run A.
+
+### Workflow Overview
+
+1. **Initialize Run A** and log training metrics.  
+2. **Initialize Run B1** (evaluation), log data, then finish it.  
+3. **Log more data to Run A**.  
+4. **Initialize Run B2**, log data, then finish it.  
+5. **Continue logging to Run A**.  
+6. **Finally finish Run A** at the end.
+
+### Sample Code
+
+```python
+import wandb
+
+def train(name: str) -> None:
+    """
+    Perform one training iteration in its own W&B run.
+
+    Using a 'with wandb.init()' block with `reinit="create_new"` ensures that
+    this training sub-run can be created even if another run (like our primary
+    tracking run) is already active.
+    """
+    with wandb.init(
+        project="my_project",
+        name=name,
+        reinit="create_new"
+    ) as run:
+        # Pretend there's a training loop here logging 'train_loss'.
+        # In a real script, you'd run your training steps inside this block.
+        run.log({"train_loss": 0.42})  # Replace with your real metric(s)
+
+def evaluate_loss_accuracy() -> (float, float):
+    """
+    Returns the current model's loss and accuracy.
+    Replace this placeholder with your real evaluation logic.
+    """
+    return 0.27, 0.91  # Example metric values
+
+# Create a 'primary' run that remains active throughout multiple train/eval steps.
+with wandb.init(
+    project="my_project",
+    name="tracking_run",
+    reinit="create_new"
+) as tracking_run:
+    # 1) Train once under a sub-run named 'training_1'
+    train("training_1")
+    loss, accuracy = evaluate_loss_accuracy()
+    tracking_run.log({"eval_loss": loss, "eval_accuracy": accuracy})
+
+    # 2) Train again under a sub-run named 'training_2'
+    train("training_2")
+    loss, accuracy = evaluate_loss_accuracy()
+    tracking_run.log({"eval_loss": loss, "eval_accuracy": accuracy})
+    
+    # The 'tracking_run' finishes automatically when this 'with' block ends.
+
+---
+
+### Key Takeaways
+
+1. **`reinit="create_new"`**  
+   Forces a new run each time you call `wandb.init()`.
+
+2. **Keep references to each run**  
+   Since `wandb.run` won’t automatically point to the newest run in `create_new` mode, store them in variables like `run_a`, `run_b1`, etc., and call `.log()` or `.finish()` on those objects.
+
+3. **Order of finishing**  
+   You can finish sub-runs whenever you want while keeping the primary run open until the very end.
+
+4.  **Always Finish Your Runs**  
+    If you don’t explicitly call `run.finish()`, your script exit may be slower, or your data may not be fully uploaded.
```

Comment:
Another reason it is important is that `create_new` will eventually become the default (though likely not soon since `wandb.run` is so embedded), and scripts that don't finish their runs explicitly will not work correctly.